dashboard/test/factories/factories.rb

```diff
@@ -1074,9 +1074,14 @@
     end
 
     factory :csp_script do
+      is_course {true}
+      sequence(:version_year) {|n| "bogus-csp-version-year-#{n}"}
+      sequence(:family_name) {|n| "bogus-csp-family-name-#{n}"}
       after(:create) do |csp_script|
         csp_script.curriculum_umbrella = Curriculum::SharedCourseConstants::CURRICULUM_UMBRELLA.CSP
         csp_script.save!
+        course_offering = CourseOffering.add_course_offering(csp_script)
+        course_offering.update!(grade_levels: '9,10,11,12')
```

Comment:
this is confusing to me -- I would expect a csp unit to be part of a course, and therefore not have a course offering, family name or version year. can you please create a different factory like `hoc_script`, `standalone_unit` or a new one which isn't already associated with being a unit-within-a-multi-unit-course?

Reply:
depending on what you need, it's possible that a csp_unit within a course would also make sense.

Reply:
This is the code that I need to test. 

```
def self.get_grade_levels(unit)
    unit.get_course_version.course_offering.grade_levels
  end
```
So, it sounds like I need to create a course, course_version _and_ course_offering that all associate with the csp_unit I want to make? I need it to be a csp_unit, because the "CSP" part is what determines the programming language which is also included in the system prompt and tested here.

Reply:
correct! I think it would look something like this: https://github.com/code-dot-org/code-dot-org/blob/1784fdf3ffa07eb36c6ca453ce338c6636ced5cf/dashboard/test/controllers/congrats_controller_test.rb#L81-L85

Reply:
updated!